lib/plug/application.ex

```diff
@@ -8,7 +8,8 @@ defmodule Plug.Application do
     Plug.Keys = :ets.new(Plug.Keys, [:named_table, :public, read_concurrency: true])
 
     children = [
-      Plug.Upload
+      Plug.Upload.Setup,
```

Comment:
Instead of having a setup process, have a regular supervisor, do the init on the regular supervisor, and have PartitionSupervisor as the child of this new supervisor. :)

Reply:
Not used to doing any init work on supervisors because OTP live upgrades would scream if the supervisor init callback has side-effects. I know that generally Elixir doesn't encourage good-old-live-code-upgrades but still, the habit is very strong 👀